src/Pausable.sol

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity >=0.8.8 <0.9.0;
+
+/**
+ * @dev Contact Module that allows children to implement logic to pause and unpause the contract.
+ * This is based on the OpenZeppelin Pausable contract but makes use of deterministic storage slots
+ * and the EVM native word size to optimize gas costs.
+ *
+ * The `whenPaused` and `whenNotPaused` modifiers are used to
+ * execute code based on the current state of the contract.
+ *
+ */
+import {Initializable} from "./libraries/external/Initializable.sol";
+
+abstract contract Pausable is Initializable {
+    /// NOTE: use uint256 to save on gas because it is the native word size of the EVM
+    /// it is cheaper than using a bool because modifying a boolean value requires an extra SLOAD
+    uint256 private constant NOT_PAUSED = 1;
+    uint256 private constant PAUSED = 2;
+
+    /**
+     * @dev Contract is not paused, functionality is unblocked
+     */
+    error RequireContractIsNotPaused();
+    /**
+     * @dev Contract state is paused, blocking
+     */
+    error RequireContractIsPaused();
+    // @dev Storage slot with the pause flag, this is managed by the `Pause` struct
+
+    struct PauseStorage {
+        uint256 _pauseFlag;
+    }
+
+    // @dev Emitted when the contract is paused
+    event Paused(bool paused);
+    event NotPaused(bool notPaused);
+
+    bytes32 public constant PAUSE_SLOT = bytes32(uint256(keccak256("Pause.pauseFlag")) - 1);
```

Comment:
We need to initialise the storage slot to `NOT_PAUSED` in an initialiser. Right now the slot will be 0 by default

Reply:
i believe it is initialized: https://github.com/wormhole-foundation/example-native-token-transfers/blob/42e0c3bcf0558dcba56a3ea5c56586b37d00888a/src/Pausable.sol#L94